### DISCUSSION TITLE
OpenVINO Execution Provider

### DISCUSSION THREAD

**Discussion by bopeng1234** [microsoft/onnxruntime-genai#1096] [Support for Targets (OS / EPs / Hardware)]
Hi there,

I am wondering if the onnxruntime-genai support [OpenVINO](https://onnxruntime.ai/docs/execution-providers/OpenVINO-ExecutionProvider.html) as an Execution Provider.

as I found in [example](https://github.com/microsoft/onnxruntime-genai/blob/main/examples/python/phi-3-tutorial.md), only provide the execution sample on CPU/CUDA/DML, no OpenVINO.

To my understanding, the OpenVINO EP now, can only run one time inference, for example tradition image classification, but can't run text generation (multi round inference and kv caching)?

Best Regards!

**Comment by RyanUnderhill** [comment]
Yes, it doesn't currently support it but we can add basic support for it (where onnxruntime uses it). I think GenAI doing CPU scoring is no problem for OpenVINO as it's CPU based.

**Comment by bopeng1234** [comment]
thanks, do we have any branch we are working on, to add openvino ep to onnxruntime-genai
